File: apps/android/app/src/main/java/app/tagscribe/models/Item.kt
package app.tagscribe.models

/**
 * Mirrors API type (camelCase). See private/specifications/mobile-apps/API-CONTRACT.md
 */
data class Item(
    val id: String,
    val type: String,
    val content: String,
    val title: String? = null,
    val highlight: String? = null,
    val caption: String? = null,
    val tags: List<String> = emptyList(),
    val categoryId: String? = null,
    val source: String? = null,
    val createdAt: String,
    val updatedAt: String,
    val archivedAt: String? = null,
)

File: apps/ios/TagScribe/Models/Item.swift
import Foundation

/// Encode/decode multiple HTTP URLs in a single `Item.content` string (newline-separated). Backward-compatible with a single URL with no newlines.
enum LinkStorage {
    static func linkLines(from content: String) -> [String] {
        content
            .split(whereSeparator: \.isNewline)
            .map { String($0).trimmingCharacters(in: .whitespaces) }
            .filter { !$0.isEmpty }
    }

    static func joinedHTTPURLs(_ lines: [String]) -> String {
        lines
            .map { $0.trimmingCharacters(in: .whitespacesAndNewlines) }
            .filter { isValidHTTPURL($0) }
            .joined(separator: "\n")
    }

    static func isValidHTTPURL(_ s: String) -> Bool {
        let t = s.trimmingCharacters(in: .whitespacesAndNewlines)
        return t.hasPrefix("http://") || t.hasPrefix("https://")
    }
}

/// One row in Add / Edit link lists (stable `id` for `ForEach` bindings).
struct EditableLinkRow: Identifiable {
    let id = UUID()
    var value: String = ""
}

/// Mirrors API type: see private/specifications/mobile-apps/API-CONTRACT.md
struct Item: Codable, Identifiable {
    let id: String
    let type: String
    let content: String
    var imageUrls: [String]?
    var title: String?
    var highlight: String?
    var caption: String?
    var tags: [String]
    var categoryId: String?
    var source: String?
    let createdAt: String
    let updatedAt: String
    var archivedAt: String?
}
